tcpproxy.go

```diff
@@ -344,8 +344,27 @@ func UnderlyingConn(c net.Conn) net.Conn {
 	}
 	return c
 }
+func tcpConn(c net.Conn) (t *net.TCPConn, ok bool) {
```

Comment:
blank line between these three new funcs, as each are longer than a line each (I think generally the rule is that funcs only touch if they're all one liners or closely related accessors?)